home/models.py

```diff
@@ -494,6 +494,44 @@ class OfflineAppPage(AbstractArticle):
     ])
 
 
+class OfflineContentIndexPage(AbstractArticle):
```

Comment:
There's a lot of duplication here. `OfflineContentIndexPage` is very similar to `OfflineAppPage` which is very similar to `AbstractArticle`. Do we need a completely new model to deal with such minor differences? Is there no better way to express the differences than to copy 30+ lines of code each time?

Reply:
refactor to remove a lot of duplication